packages/services/api/src/modules/schema/providers/schema-manager.ts

```diff
@@ -99,46 +102,60 @@ export class SchemaManager {
 
   @traceFn('SchemaManager.compose', {
     initAttributes: input => ({
-      'hive.target.id': input.targetId,
-      'hive.organization.id': input.organizationId,
-      'hive.project.id': input.projectId,
+      'hive.organization.slug': input.target?.bySelector?.organizationSlug,
+      'hive.project.slug': input.target?.bySelector?.projectSlug,
+      'hive.target.slug': input.target?.bySelector?.targetSlug,
+      'hive.target.id': input.target?.byId ?? undefined,
       'input.only.composable': input.onlyComposable,
       'input.services.count': input.services.length,
     }),
   })
-  async compose(
-    input: TargetSelector & {
-      onlyComposable: boolean;
-      services: ReadonlyArray<{
-        sdl: string;
-        url?: string | null;
-        name: string;
-      }>;
-    },
-  ) {
+  async compose(input: {
+    onlyComposable: boolean;
+    services: ReadonlyArray<{
+      sdl: string;
+      url?: string | null;
+      name: string;
+    }>;
+    target: GraphQLSchema.TargetReferenceInput | null;
+  }) {
     this.logger.debug('Composing schemas (input=%o)', lodash.omit(input, 'services'));
+
+    const selector = await this.idTranslator.resolveTargetReference({
+      reference: input.target ?? null,
+      onError() {
+        throw new InsufficientPermissionError('schema:compose');
```

Comment:
Could other errors happen?

Reply:
Internally a few things can happen, but we catch these log them, and then treat all of those as "InsufficientPermissionsError" to the requester.  E.g. if someone provides `99999` as the ID, we don't throw a "please send an UUID" error, but this one instead.